### ISSUE TITLE
EXPOSE with variables doesn't work

### ISSUE THREAD

**Issue by ttres** [hadolint/hadolint#53]
The Dockerfile validation fails when the ports arguments to the EXPOSE command are declared as environment variables. Take this dummy, useless code as example:

``` Docker
FROM alpine:latest
ENV PORT=8080
EXPOSE $PORT
```

This is the output:

``` Bash
$ docker run --rm -i lukasmartinelli/hadolint < Dockerfile
"/dev/stdin" (line 3, column 8):
unexpected '$'
expecting space, "\t", "ONBUILD", "FROM", "COPY", "RUN", "WORKDIR", "ENTRYPOINT", "VOLUME", "EXPOSE", "ENV", "ARG", "USER", "LABEL", "STOPSIGNAL", "CMD", "MAINTAINER", "ADD", "#" or end of input
```

But the image inspection show that the correct port was exposed:

``` JSON
$ docker build --tag=sample -f Dockerfile .
$ docker inspect sample | grep -A2 Exposed
            "ExposedPorts": {
                "8080/tcp": {}
            },
```

Declaring variables with the exposed ports allow for reuse inside the Dockerfile.

**Comment by lukasmartinelli** [comment]
> The Dockerfile validation fails when the ports arguments to the EXPOSE command are declared as environment variables. Take this dummy, useless code as example:

I don't support variables anywhere right now.
Perhaps I could just ad another preprocessing pipeline that replaces the env vars with the declared values.

1. Scan through file once
2. Build AST
3. Filter out ENV statements
4. Replace `$ENV` in file with `$ACTUAL_VALUE`
..
1. And then start parsing for the real thing (scan through file, build ast, do linting)

In a language parser this stuff is probably bad due to performance issues but here it is not so critical.

**Comment by ttres** [comment]
> In a language parser this stuff is probably bad due to performance issues but here it is not so critical.

I understand the issue.

Your approach sounds good to me.

**Comment by StevenACoffman** [comment]
Is there a pull request where this work is being done?
